libnmstate/nm/connection.py

```diff
@@ -21,26 +21,67 @@
 from . import nmclient
 
 
-def create_profile(settings):
-    con_profile = nmclient.NM.SimpleConnection.new()
-    for setting in settings:
-        con_profile.add_setting(setting)
+class ConnectionProfile(object):
+
+    def __init__(self, profile=None):
+        self._con_profile = profile
+        self._nmclient = nmclient.client()
+        self._mainloop = nmclient.mainloop()
+
+    def create(self, settings):
+        self._con_profile = nmclient.NM.SimpleConnection.new()
+        for setting in settings:
+            self._con_profile.add_setting(setting)
+
+    def import_by_device(self, nmdev):
+        self._con_profile = None
+        ac = get_device_active_connection(nmdev)
+        if ac:
+            self._con_profile = ac.props.connection
+
+    def import_by_id(self, con_id):
+        self._con_profile = None
+        if con_id:
+            self._con_profile = self._nmclient.get_connection_by_id(con_id)
+
+    def update(self, profile):
```

Comment:
This should take a ConnectionProfile object. Otherwise the abstraction is broken.

Reply:
I'm not convinced there is an abstraction problem here.
I am ok with your suggestion, but it depends if it is easy to use at the caller.
I mean, if we allow to input in the constructor the profile, then it is easy, otherwise it will be a lot of steps for the caller.

Reply:
Usage concurs with you for now, changing this.